### ISSUE TITLE
Add <app-media-shape-detection> element

### ISSUE THREAD

**Issue by FluorescentHallucinogen** [PolymerElements/app-media#7]
@cdata First of all, thank you so much for all your work! `<app-media>` elements are awesome! :thumbsup:

FYI, I've added `<app-media>` elements into my [`progressive-elements`](https://webcomponents.org/collection/FluorescentHallucinogen/progressive-elements) collection that contains web components for modern features/APIs (i.e. building blocks for progressive web apps).

What about Shape Detection API web component (e.g. `<app-media-shape-detection>`)? :wink:

You can read more about Shape Detection API in @PaulKinlan's blog:

https://paul.kinlan.me/face-detection/
https://paul.kinlan.me/barcode-detection/
https://paul.kinlan.me/detecting-text-in-an-image/

TL;DR: Shape Detection API is a Web API for QR/barcode/text/face detection in live or still images. See use cases: https://github.com/WICG/shape-detection-api#use-cases-camera

In the near future, I plan to develop web components for [Google Cloud Vision API](https://cloud.google.com/vision/), [Google Cloud Speech API](https://cloud.google.com/speech/), [Google Cloud Translate API](https://cloud.google.com/translate/).

It helps easily and quickly develop mind-blowing progressive web apps similar to @dmotz's [Thing Translator](https://aiexperiments.withgoogle.com/thing-translator) using web components only. :fire:

Also I'm thinking about web component for [NTechLab FindFace API](https://findface.pro/en/), but that's a different story… :stuck_out_tongue_winking_eye:

**Comment by cdata** [comment]
Shape Detection is still highly experimental (even more so than Image Capture). Currently there are bugs in the only native implementation (e.g., memory leaks) and also it's possible that the API will change substantially as the proposal advances through the standards process.

All that said, we're extremely excited about Shape Detection and will definitely add it to this collection in a future release. In the mean time, consider experimenting with your own wrapper that makes use of these elements.
